python/ray/util/spark/cluster_init.py

```diff
@@ -481,9 +490,25 @@ def _setup_ray_cluster(
 
     if ray_temp_root_dir is None:
         ray_temp_root_dir = start_hook.get_default_temp_dir()
-    ray_temp_dir = os.path.join(
-        ray_temp_root_dir, f"ray-{ray_head_port}-{cluster_unique_id}"
-    )
+    if global_mode_enabled():
+        # It reads from environment variable RAY_ADDRESS, then check
+        # /local_disk0/tmp/ray/ray_current_cluster to see if the address
+        # is available, if yes then a global ray cluster is running, otherwise
+        # it will start a new glbal ray cluster.
+        ray_addr = canonicalize_bootstrap_address_or_die("auto", get_user_temp_dir())
```

Comment:
We need add verification in `setup_ray_cluster` to disallow user to set the `ray_temp_root_dir` argument.

Reply:
Pls also update doc of `setup_ray_cluster` to tell user how to enable global mode (databricks only),
and list risks and limitations for global mode:
 - limitations: only allow one global mode ray cluster at a time
 - risks: If user forgot shutdown global mode ray cluster, ray head node / worker nodes will keep running, and it will block databricks cluster auto-termination.